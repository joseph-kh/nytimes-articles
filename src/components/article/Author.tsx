import React from "react";
import { TextProps, StyleProp, TextStyle, StyleSheet } from "react-native";
import ItalicText from "@/components/ui/texts/ItalicText";
import COLORS from "@/theme/colors";

type Props = {
  author: string;
  style?: StyleProp<TextStyle>;
};

const Author: React.FC<Props & TextProps> = (props) => {
  const { author, style } = props;

  return (
    <ItalicText
      {...props}
      style={StyleSheet.flatten([styles.textStyle, style])}
    >
      {author}
    </ItalicText>
  );
};

export default Author;

const styles = StyleSheet.create({
  textStyle: {
    color: COLORS.gray,
  },
});
