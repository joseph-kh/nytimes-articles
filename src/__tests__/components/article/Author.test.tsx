import React from "react";
import renderer from "react-test-renderer";
import Author from "@/components/article/Author";

describe("Author Component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Author author="John Doe" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("applies custom styles", () => {
    const customStyle = { fontSize: 16 };
    const tree = renderer
      .create(<Author author="John Doe" style={customStyle} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
