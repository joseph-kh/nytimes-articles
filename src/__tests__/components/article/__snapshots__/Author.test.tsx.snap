// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Component applies custom styles 1`] = `
<Text
  author="John Doe"
  style={
    {
      "color": "#7e7e7e",
      "fontFamily": "Italic",
      "fontSize": 16,
    }
  }
>
  John Doe
</Text>
`;

exports[`Author Component renders correctly 1`] = `
<Text
  author="John Doe"
  style={
    {
      "color": "#7e7e7e",
      "fontFamily": "Italic",
      "fontSize": 21,
    }
  }
>
  John Doe
</Text>
`;
